src/ingo_hash/mod.rs

```diff
@@ -1,3 +1,6 @@
-pub mod hash_hw_code;
+mod hash_hw_code;
 pub mod poseidon_api;
 pub mod utils;
+
+pub use poseidon_api::*;
+pub use utils::*;
```

Comment:
Whats the purpose of keeping both of these sets if both are public?

Reply:
It has different meanings
`mod` using for internal interface of library
 `use` is used to re-export a function inside another module. This allows us to present an external interface that may not directly map to internal code organization